docs/book/src/development/development.md

```diff
@@ -42,3 +42,103 @@ After generating `infrastructure-components.yaml`, replace the `us.gcr.io/k8s-ar
 ## Developing with Tilt
 
 We have support for using [Tilt](https://tilt.dev/) for rapid iterative development. Please visit the [Cluster API documentation on Tilt](https://master.cluster-api.sigs.k8s.io/developer/tilt.html) for information on how to set up your development environment. 
+
+## Running E2E tests locally
+
+You can run the E2E tests locally with:
+
+```bash
+make test-e2e OPENSTACK_CLOUD_YAML_FILE=/path/to/clouds.yaml OPENSTACK_CLOUD=mycloud
+```
+
+where `mycloud` is an entry in `clouds.yaml`.
+
+The E2E tests:
+* Build a CAPO image from the local working directory
+* Create a kind cluster locally
+* Deploy downloaded CAPI, and locally-build CAPO to kind
+* Create an e2e namespace per-test on the kind cluster
+* Deploy cluster templates to the test namespace
+* Create test clusters on the target OpenStack
+
+### Support for clouds using SSL
+
+If your cloud requires a cacert you must also pass this to make via `OPENSTACK_CLOUD_CACERT_B64`, i.e.:
+
+```bash
+make test-e2e OPENSTACK_CLOUD_YAML_FILE=/path/to/clouds.yaml OPENSTACK_CLOUD=my_cloud \
+              OPENSTACK_CLOUD_CACERT_B64=$(base64 -w /path/to/mycloud-ca.crt)
+```
+
+CAPO deployed in the local kind cluster will automatically pick up a `cacert` defined in your `clouds.yaml` so you will see servers created in OpenStack without specifying `OPENSTACK_CLOUD_CACERT_B64`. However, the cacert won't be deployed to those servers, so kubelet will fail to start.
```

Comment:
not sure I fully understand this ,the cacert is mostly used for openstack interaction defined in clouds.yaml
so why it's related to kubelet?

Reply:
We also need to pass the cacert into the cluster so the cloud provider [1] can talk to OpenStack. If the cloud provider fails to come up then the Node will not become ready and the tests fail.

[1] kubelet in-tree provider, or CCM for external cloud provider

Reply:
ok, OCCM need this cacert, thanks for the reminder